src/donor_reporting_portal/api/views/sharepoint.py

```diff
@@ -30,38 +32,47 @@ class SharePointGroupViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = SharePointGroupSerializer
 
 
-class DonorReportingViewSet:
+class DRPViewSet:
     permission_classes = ((DonorPermission | PublicLibraryPermission),)
 
 
-class DRPSharePointSettingsRestViewSet(DonorReportingViewSet, SharePointSettingsRestViewSet):
+class DRPSharePointSettingsRestViewSet(DRPViewSet, SharePointSettingsRestViewSet):
     serializer_class = DRPSharePointSettingsSerializer
 
 
-class DRPSharePointSettingsCamlViewSet(DonorReportingViewSet, SharePointSettingsCamlViewSet):
+class DRPSharePointSettingsCamlViewSet(DRPViewSet, SharePointSettingsCamlViewSet):
     serializer_class = DRPSharePointSettingsSerializer
 
 
-class DRPSharePointUrlRestViewSet(DonorReportingViewSet, SharePointUrlRestViewSet):
+class DRPSharePointUrlRestViewSet(DRPViewSet, SharePointUrlRestViewSet):
     serializer_class = DRPSharePointUrlSerializer
 
 
-class DRPSharePointUrlCamlViewSet(DonorReportingViewSet, SharePointUrlCamlViewSet):
+class DRPSharePointUrlCamlViewSet(DRPViewSet, SharePointUrlCamlViewSet):
     serializer_class = DRPSharePointUrlSerializer
 
 
-class DRPSharePointUrlFileViewSet(DonorReportingViewSet, SharePointUrlFileViewSet):
+class DRPSharePointUrlFileViewSet(DRPViewSet, SharePointUrlFileViewSet):
     pass
 
 
-class DRPSharePointSettingsFileViewSet(DonorReportingViewSet, SharePointSettingsFileViewSet):
+class DRPSharePointSettingsFileViewSet(DRPViewSet, SharePointSettingsFileViewSet):
     pass
 
 
-class DRPSharepointSearchViewSet(DonorReportingViewSet, SharePointSearchViewSet):
+class DRPSharepointSearchViewSet(SharePointSearchViewSet):
     prefix = 'DRP'
     serializer_class = DRPSharePointSearchSerializer
 
+    def is_public(self):
```

Comment:
Finding this method a little confusing.

You have the keys hard coded here, why not just use them?
e.g.

`source_id in [settings.DRP_SOURCE_IDS['thematic_internal'], settings.DRP_SOURCE_IDS['thematic_external']]`

and

`source_id in [settings.DRP_SOURCE_IDS['internal'], settings.DRP_SOURCE_IDS['pool_internal']]`

I'd also just return after the after setting `public` variable if it is `True`, no need to perform other logic.

Also have an `if self.request.user.username.endswith('@unicef.org'):` wrapping the internal logic. I'm finding the `internal` variable name very confusing with the `is_public` method :)

And default with `False`

Reply:
@reinbach does it look more clear?